content/en/docs/concepts/architecture/nodes.md

```diff
@@ -173,18 +173,30 @@ to be unreachable. (The default timeouts are 40s to start reporting
 ConditionUnknown and 5m after that to start evicting pods.) The node controller
 checks the state of each node every `--node-monitor-period` seconds.
 
-In versions of Kubernetes prior to 1.13, NodeStatus is the heartbeat from the
-node. Node lease feature is enabled by default since 1.14 as a beta feature
-(feature gate `NodeLease`, [KEP-0009](https://github.com/kubernetes/enhancements/blob/master/keps/sig-node/0009-node-heartbeat.md)).
-When node lease feature is enabled, each node has an associated `Lease` object in
-`kube-node-lease` namespace that is renewed by the node periodically, and both
-NodeStatus and node lease are treated as heartbeats from the node. Node leases
-are renewed frequently while NodeStatus is reported from node to master only
-when there is some change or enough time has passed (default is 1 minute, which
-is longer than the default timeout of 40 seconds for unreachable nodes). Since
-node lease is much more lightweight than NodeStatus, this feature makes node
-heartbeat significantly cheaper from both scalability and performance
-perspectives.
+#### Heartbeats
+
+Heatbeats, sent by Kubernetes nodes, help determining their state.
+There are two forms of hearbeats: updates of `NodeStatus` an
```

Comment:
@wojtek-t , Some more suggestions+nits.
line 171- 174: Somewhat related: these lines seem to be a repeat of lines 187 -190?
line 178: Spelling, wording: Heartbeats, sent by Kubernetes nodes [to the control plane], help determine the health [or availability] of a node.
line 179: Spelling, wording: There are two forms of heartbeats: updates containing `NodeStatus` and updates containing the (Lease object){ ...].
line 181: in the `kube-node-lease` ...
line 191: Wording to make concise: The `kubelet` updates the `Lease` object every 10 seconds (the default update interval).
line 195 - 197: Possibly move content, reduce, and merge with line 183 (@sftim , comments on including this statement?):
`Lease` is a lightweight resource, which improves the performance
of the node heartbeats as the cluster scales.

Reply:
> line 171- 174: Somewhat related: these lines seem to be a repeat of lines 187 -190?

I disagree - 171-174 is talking what nodecontroller is doing based on signals it get
lines 187-190 is discussing what those signals are

> line 179: Spelling, wording: There are two forms of heartbeats: updates containing NodeStatus and updates containing the (Lease object){ ...].

Those aren't updates that contain something - in the REST api these are updates of objects.


Other comments should addressed.

Reply:
@wojtek-t , OK. Almost there.

line 178: pls. take out square brackets -- I intended you to choose the words that fit best (where are the heartbeats sent? to the control plane, _availability_ vs. _health_):
Heartbeats, sent by Kubernetes nodes, help determine the availability of a node.
line 180: s/hearbeats/heartbeats/
line 180: If this makes sense, replace _an- with _and the_: updates of `NodeStatus` and the ...
line 193: Reverting back to two sentences, as it seems easier to read
The `kubelet` creates and then updates the `Lease` object every 10 seconds (the default update interval).
`Lease` updates occur independently from the `NodeStatus` updates.

Reply:
Thanks a lot for patience. Done.